File: powerUtils.cpp
/**
* Bill Siever
* 2024-03-07 Initial Version
*
* Development environment specifics:
* Written in Microsoft PXT
*
* This code is released under the [MIT License](http://opensource.org/licenses/MIT).
* Please review the LICENSE.md file included with this example. If you have any questions 
* or concerns with licensing, please contact techsupport@sparkfun.com.
* Distributed as-is; no warranty is given.
*/

#include "pxt.h"
#include "MicroBit.h"

using namespace pxt;

namespace powerUtils { 
    //%
    int usbPower() {
        MicroBitPowerSource source = uBit.power.getPowerSource();
        return source == PWR_USB_ONLY || source == PWR_USB_AND_BATT;
    }

    //%
    int powerConsumption() {
        return uBit.power.getPowerConsumption();
    }
}

File: powerUtils.ts


//% color=#000000 
//% icon="\uf1e6"
//% block="Power Utilities"
//% group="micro:bit (V2)"
namespace powerUtils {

    

    //% block="USB Power"
    //% shim=powerUtils::usbPower
    //% group="micro:bit (V2)"
    export function usbPower() : boolean {
        // Per https://github.com/microsoft/pxt-microbit/issues/4292
        return true;
    }

    //% block="instantaneous power consumption"
    //% shim=powerUtils::powerConsumption
    //% group="micro:bit (V2)"
    export function powerConsumption() : number {
        // Per https://github.com/microsoft/pxt-microbit/issues/4292
        return 0;
    } 
}
